modules/vaos/spec/services/concerns/token_authentication_spec.rb

```diff
@@ -0,0 +1,149 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require_relative '../../../app/services/concerns/token_authentication'
+require_relative '../../../app/services/concerns/jwt_wrapper'
+require 'common/client/concerns/monitoring'
+
+# Base class for testing that implements perform
+# This is used so we don't have to implement all other functions of the service session class.
+class TestServiceBase
```

Comment:
Note: This one gave me a bit of trouble as there is a lot going on on that perform method call, so it is mocked here so as to not test all that code that is of no concern to this concern!